src/ext.rs

```diff
@@ -0,0 +1,121 @@
+use core::{marker::PhantomData, task::Poll};
+
+use crate::{
+    api::{reply, request},
+    backend::Backend,
+    client::{ClientError, ClientImplementation, FutureResult, PollClient},
+    error::Error,
+    platform::{Platform, Syscall},
+    postcard_deserialize, postcard_serialize_bytes,
+    service::ServiceResources,
+    types::ClientContext,
+};
+
+use serde::{de::DeserializeOwned, Serialize};
+
+pub trait Extension {
+    type Request: DeserializeOwned + Serialize;
```

Comment:
I think we want to have `type Request<'a>: Deserialize<'a> + Serialize`. Same for the Response type.

Otherwise we risk having high stack usage: Say a command comes from the host. The app would have to parse it's relevant data into a `Request` that looks like `struct Request { field1: Bytes<1000>, field2…}` which takes 1000 bytes on the stack. The same thing then happens in the backend. The same thing also happens with the reply.

By using the lifetimes I suggested we can instead use `struct Request<'a> { field1: &'a [u8], field2: …}`, which requires very little stack space.

Reply:
I realize this doesn't work with the current `Poll` trait that returns an owned reply. It does seem inefficient to always move the request/replies, given that they're pretty large.

I think returning references and let the client `clone` when necessary would be more efficient.